lib/ansible/modules/cloud/amazon/aws_netapp_cvs_active_directory.py

```
@@ -0,0 +1,269 @@
+            data = None
+            response, error = self.restApi.delete(api, data)
+            if not error:
+                return response
+            else:
+                self.module.fail_json(msg=response['message'])
+
+        else:
+            self.module.fail_json(msg="Active Directory does not exist")
+
+    def update_activedirectory(self, activedirectoryId, updated_activedirectory):
+        # Update ActiveDirectory
+        api = 'Storage/ActiveDirectory/' + activedirectoryId
+        data = {
+            "region": self.parameters['region'],
+            "DNS": updated_activedirectory['DNS'],
+            "domain": updated_activedirectory['domain'],
+            "username": updated_activedirectory['username'],
+            "password": updated_activedirectory['password'],
+            "netBIOS": updated_activedirectory['netBIOS']
+        }
+
+        response, error = self.restApi.put(api, data)
+        if not error:
+            return response
+        else:
+            self.module.fail_json(msg=response['message'])
+
+    def apply(self):
+        """
+        Perform pre-checks, call functions and exit
+        """
+        modify = False
+        activeDirectoryId = self.get_activedirectoryId()
+        current = self.get_activedirectory(activeDirectoryId)
+        cd_action = self.na_helper.get_cd_action(current, self.parameters)
+
+        if current and self.parameters['state'] != 'absent':
+            keys_to_check = ['DNS', 'domain', 'username', 'password', 'netBIOS']
+            updated_active_directory, modify = self.na_helper.compare_and_update_values(current, self.parameters, keys_to_check)
+
+            if modify is True:
+                self.na_helper.changed = True
+                if 'domain' in self.parameters and self.parameters['domain'] is not None:
+                    ad_exists = self.get_activedirectory(updated_active_directory['domain'])
+                    if ad_exists:
```

I would report an error here, as we did not complete the action.

But this change can be delayed.